lms/djangoapps/lti_provider/signature_validator.py

```diff
@@ -0,0 +1,244 @@
+    # hashed. By default, the oauthlib library rejects any URLs that do not
+    # use HTTPS. We turn this behavior off in order to allow edX to run without
+    # SSL in development mode. When the platform is deployed and running with
+    # SSL enabled, the URL passed to the signature verifier must start with
+    # 'https', otherwise the message signature would not match the one generated
+    # on the platform.
+    enforce_ssl = False
+
+    def check_client_key(self, key):
+        """
+        Verify that the key supplied by the LTI consumer is valid for an LTI
+        launch. This method is only concerned with the structure of the key;
+        whether the key is associated with a known LTI consumer is checked in
+        validate_client_key. This method signature is required by the oauthlib
+        library.
+
+        :return: True if the client key is valid, or False if it is not.
+        """
+        return key is not None and 0 < len(key) <= 32
+
+    def check_nonce(self, nonce):
+        """
+        Verify that the nonce value that accompanies the OAuth signature is
+        valid. This method is concerned only with the structure of the nonce;
+        the validate_timestamp_and_nonce method will check that the nonce has
+        not been used within the specified time frame. This method signature is
+        required by the oauthlib library.
+
+        :return: True if the OAuth nonce is valid, or False if it is not.
+        """
+        return nonce is not None and 0 < len(nonce) <= 64
+
+    def validate_timestamp_and_nonce(self, client_key, timestamp, nonce,
+                                     request, request_token=None,
+                                     access_token=None):
+        """
+        Verify that the request is not too old (according to the timestamp), and
+        that the nonce value has not been used already within the period of time
+        in which the timestamp marks a request as valid. This method signature
+        is required by the oauthlib library.
+
+        :return: True if the OAuth nonce and timestamp are valid, False if they
+        are not.
+        """
+        return True
+
+    def validate_client_key(self, client_key, request):
+        """
+        Ensure that the client key supplied with the LTI launch is on that has
+        been generated by our platform, and that it has an associated client
+        secret.
+
+        :return: True if the key is valid, False if it is not.
+        """
+        return LtiConsumer.objects.filter(key=client_key).count() == 1
+
+    def get_client_secret(self, client_key, request):
+        """
+        Fetch the client secret from the database. This method signature is
+        required by the oauthlib library.
+
+        :return: the client secret that corresponds to the supplied key if
+        present, or None if the key does not exist in the database.
+        """
+        try:
+            return LtiConsumer.objects.get(key=client_key).secret
```

Comment:
Generally I prefer to call methods as opposed to attributes. This allows you to change the name of attributes or how they are represented, without changing associated calls.

What I mean is if this is, `.get_secret()` then if later on, the secret has to be transformed or something, you can just change the logic of `get_secret()`'s implementation ,rather than change all the places you referenced the attribute `.secret`.